### ISSUE TITLE
Release notes for review

### ISSUE THREAD

**Issue by itsNintu** [webstudio-is/webstudio#2578]
# Release notes for version 0.112.0 - 0.114.0

### New Features:

- **Automatic sitemap & robots generation:**
To improve your webpage's discoverability, our platform now auto-generates both a sitemap.xml and robots.txt for all published sites. This feature simplifies and enhances search engine indexing.

- **Site settings dialog:**
We've introduced a user-friendly dialog in the builder menu that allows you to conveniently configure site settings.

- **Site name field in site settings:**
Our new site name field in Site Settings improves SEO and user experience by letting you set a unique site name. This ensures your site is easily identifiable in search results and browser bookmarks.

- **Favicon upload in site settings:**
For a more personalized touch, a favicon upload field is now available in site settings. You can directly upload your desired favicon, reinforcing brand consistency across all tabs.

- **Custom code field in site settings:**
You now have even more customization control with our new custom code field in site settings. Add custom scripts or stylesheets to create site-wide changes, track diverse data, or integrate third-party applications.

- **Google Search Preview block in page settings:**
Get previews of how your page might appear in Google Search results with our convenient Google Search Preview block under page settings.

- **Customize page title in page settings:**
You can now specifically customize the page title in Page Settings, offering improved control over SEO and how your page is identified in search results and tabs.

- **Customize page description in page settings:**
Page Settings now include an option for setting a customized page description. This feature provides a summarized description of your webpage for search engine results and improves SEO.

- **Exclude pages from search/index in page settings:**
Want to keep some pages private? You can exclude specific pages from being indexed by search engines.

- **Customize OG featured image of each page in page settings:**
Improve your social sharing visuals by customizing the Open Graph (OG) featured image for each page directly from the page settings.

- **Add social card support when sharing to social platforms in page settings:**
Social card metadata support for all major platforms. 

- **Custom metadata for pages in page settings:**
Add additional meta tags to each page.

### Features:

- **New UI element to control CSS transitions:**
We've made managing the CSS transition-property simpler by adding a user-friendly interface component to the transition section.

- **Unified scroll to textarea, allow autogrow:**
We have improved the text area component by enabling it to auto-resizes dynamically as you input text. You can set a limit to the maximum number of rows to prevent infinite resizing. This can be seen in the settings panel which now uses these new auto-growing text areas.

- **Enable AI for everyone:**
Webstudio AI has now been enabled for all users! 

### Fixes:

- **Resolved issue with images not displaying**:
Fixed a problem with images being ratelimited and not displayed correctly.

- **Resolved issue with image stretch in page settings:**
Fixed a problem where the 'No Image' placeholder image on the page settings was being distorted/stretched.

- **Resolved project outdated issue when switching to preview and using link anchors**: 
We have fixed an issue that would trigger an error saying "The project is outdated. Synchronization was incomplete when the project was opened". This issue occurred when loading outdated project versions during regular user work.
  
- **Resolved version mismatch error for post-transaction issues:**
We have fixed several issues related to disconnections and incorrect version mismatch errors.

- **Resolved URL misformatting issue in Page Setting:**
We have fixed an issue where the link in Page Setting displayed the internal name/subdomain without a proper extension.

- **Resolved Wrong zIndex on favicon upload:**
A fix has been issued to resolve the incorrect layering of the favicon during the upload process.

- **Resolved Alignment of drop-down menu under user avatar fixed:**
Fixed the alignment of the drop-down menu appearing under the user avatar.

- **Resolved textarea scrollbar in Safari:**
Implementation of a fix to prevent the appearance of an unnecessary scrollbar in text-area elements when viewed in Safari.

- **Resolved Resize canvas with breakpoints hotkeys:**
Breakpoints shortcuts (numbers 1, 2, 3, etc.) have been fixed and now properly resize the canvas.

- **Resolved Incorrect defaultValue in text prop:**
Fixed a bug where the text property used uncontrolled defaultValue causing conflicts with history management.

### Features:
- **Added new variables panel:**
We've added a panel specifically for maintaining variables. Current features include create/edit/delete.

- **Added new basic code editor:**
Now enjoy the presence of a built-in code editor with variables completion support, which aids in the UI design process. The editor acknowledges the re-using of variable names and offers auto-completion through the use of $.

- **Improved normalization of variable names:**
We have made it easier for you to create valid variable names. Now, you can provide any name and our system will generate a proper variable name without unnecessary restrictions. This improvement ensures that you can safely use variable names to reference expressions without breaking projects.

- **Enhanced support for binding props to variables:**
We've expanded the variables panel user interface to allow you to bind expressions to props. By doing so, you can now reference and compute values from variables, adding more flexibility to your creative workflow. Our expressions editor supports auto-completion with $, or you can simply select a variable from the list to generate the expression automatically.

# New Contributors
A massive welcome and thank you to our newest community collaborators: 
@ccthecode made their first contribution in https://github.com/webstudio-is/webstudio/pull/2503
@faisal004 made their first contribution in https://github.com/webstudio-is/webstudio/pull/2528

**Comment by itsNintu** [comment]
@kof for review

**Comment by kof** [comment]
@itsNintu transitions are behind feature flag yet, so not available to the users

**Comment by kof** [comment]
Categories: builder updates vs SEO don't make sense, all those seo settings are inside builder

**Comment by kof** [comment]
This is the prompt I used to generate changelog with GPT from the current changelogs, this has to be mostly automated.

```
   You are writing a software changelog: 
   1. Group all points by the prefix, use that prefix as a title, don't lose any changes, use each title only once
   2. Simplify the wording in a way that
      - makes it easier to read
      - fix the grammar
      -  keep the links as-is
   
   Example:
   ## Feat
   Implemented Rate Limiting by @istarkov (https://github.com/webstudio-is/webstudio/pull/2393)
   
   End of Example
   {changes}
   ```

**Comment by kof** [comment]
This is the changelog I generated this way: https://github.com/webstudio-is/webstudio/releases/tag/0.105.0

**Comment by kof** [comment]
If you are not happy with it, let me know, we need to find a format that works well and automate  this.

**Comment by itsNintu** [comment]
Thanks @kof 

I started with a similar prompt and gradually refined it to get something that can be understood by non technical individuals.  
The release notes you shared above are too complex for normal humans to understand. 

With some training examples and the addition to your prompt we can 100% automate this. I'll get it setup over the weekend and record a video walkthrough too.

For larger releases there might be value in grouping into categories like seo/style/variables/animations etc, but otherwise I agree let's keep it simple.


We should include functionality that is built but not yet released. It will keep users engaged with these updates and keep them returning to see the next release notes to see if that thing they wanted has been released yet. I just need to make sure it's very clear this will be part of an upcoming release.

**Comment by kof** [comment]
Well, I don't feel your writeup is good enough because of:
1. misleading categories: SEO & Social Sharing vs Builder Updates
2. too much manual editing, we need to automate
3. not available features to the user need to be separated or marked as such (we need to tag them somehow in PR titles), otherwise this will lead to users asking where is X

**Comment by kof** [comment]
try to create what we think is right with 0 manual editing, only prompts
